File: algo/lc.125.cc
class Solution {
public:
    bool isPalindrome(string s) {
        if (0 == s.size())
            return true;
        for (int i = 0; i < s.size(); i++) {
            s[i] = tolower(s[i]);
        }
        int curl = 0, curr = s.size()-1;
        while (curl < curr) {
            if (!isalpha(s[curl]) && !isdigit(s[curl])) {
                curl++;
            } else if (!isalpha(s[curr]) && !isdigit(s[curr])) {
                curr--;
            } else if (s[curl] != s[curr]) {
                return false;
            } else { // s[curl] == s[curr]
                curl++; curr--;
            }
        }
        return true;
    }
};

File: algo/lc.125.py
class Solution:
    def isAlphaNumeric(self, c: str):
        # len(c) must be 1
        return c.isalpha() or c.isdigit()
    def isPalindrome(self, s: str) -> bool:
        newstr = [x for x in s.lower() if self.isAlphaNumeric(x)]
        return newstr == newstr[::-1]
